File: app/src/main/java/spit/spdagger2/component/NetComponent.java
package spit.spdagger2.component;

import com.squareup.okhttp.OkHttpClient;

import android.content.SharedPreferences;

import javax.inject.Singleton;

import dagger.Component;
import retrofit.Retrofit;
import spit.spdagger2.module.AppModule;
import spit.spdagger2.module.NetModule;

@Singleton
@Component(modules={AppModule.class, NetModule.class})
public interface NetComponent {
    // downstream components need these exposed
    Retrofit retrofit();
    OkHttpClient okHttpClient();
    SharedPreferences sharedPreferences();
}

File: app/src/main/java/spit/spdagger2/kotlinmvpdagger2/di/component/NetComponent.kt
package spit.spdagger2.kotlinmvpdagger2.di.component


import android.content.SharedPreferences
import com.squareup.okhttp.OkHttpClient
import dagger.Component
import retrofit.Retrofit
import spit.spdagger2.kotlinmvpdagger2.BaseApp
import spit.spdagger2.kotlinmvpdagger2.di.modules.AppModule
import spit.spdagger2.kotlinmvpdagger2.di.modules.NetModule
import spit.spdagger2.kotlinmvpdagger2.ui.main.MainActivity
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class, NetModule::class))
interface NetComponent {

//    fun inject(application: MainActivity)

    // downstream components need these exposed

   /* fun retrofit(): Retrofit?

    fun okHttpClient(): OkHttpClient?

    fun sharedPreferences(): SharedPreferences?*/
}